src/FEValues/FunctionValues.jl

```diff
@@ -0,0 +1,78 @@
+# Helpers to get the correct types for FunctionValues for the given function and, if needed, geometric interpolations. 
+struct SInterpolationDims{rdim,sdim} end
+struct VInterpolationDims{rdim,sdim,vdim} end
+function InterpolationDims(::ScalarInterpolation, ip_geo::VectorizedInterpolation{sdim}) where sdim
+    return SInterpolationDims{getdim(ip_geo),sdim}()
+end
+function InterpolationDims(::VectorInterpolation{vdim}, ip_geo::VectorizedInterpolation{sdim}) where {vdim,sdim}
+    return VInterpolationDims{getdim(ip_geo),sdim,vdim}()
+end
+
+typeof_N(::Type{T}, ::SInterpolationDims) where T = T
+typeof_N(::Type{T}, ::VInterpolationDims{<:Any,dim,dim}) where {T,dim} = Vec{dim,T}
+typeof_N(::Type{T}, ::VInterpolationDims{<:Any,<:Any,vdim}) where {T,vdim} = SVector{vdim,T} # Why not ::Vec here?
+
+typeof_dNdx(::Type{T}, ::SInterpolationDims{dim,dim}) where {T,dim} = Vec{dim,T}
+typeof_dNdx(::Type{T}, ::SInterpolationDims{<:Any,sdim}) where {T,sdim} = SVector{sdim,T} # Why not ::Vec here?
+typeof_dNdx(::Type{T}, ::VInterpolationDims{dim,dim,dim}) where {T,dim} = Tensor{2,dim,T}
+typeof_dNdx(::Type{T}, ::VInterpolationDims{<:Any,sdim,vdim}) where {T,sdim,vdim} = SMatrix{vdim,sdim,T} # If vdim=sdim!=rdim Tensor would be possible...
+
+typeof_dNdξ(::Type{T}, ::SInterpolationDims{dim,dim}) where {T,dim} = Vec{dim,T}
+typeof_dNdξ(::Type{T}, ::SInterpolationDims{rdim}) where {T,rdim} = SVector{rdim,T} # Why not ::Vec here?
+typeof_dNdξ(::Type{T}, ::VInterpolationDims{dim,dim,dim}) where {T,dim} = Tensor{2,dim,T}
+typeof_dNdξ(::Type{T}, ::VInterpolationDims{rdim,<:Any,vdim}) where {T,rdim,vdim} = SMatrix{vdim,rdim,T} # If vdim=rdim!=sdim Tensor would be possible...
+
+struct FunctionValues{IP, N_t, dNdx_t, dNdξ_t}
```

Comment:
Since we already discuss naming in this PR I put it here: 

I just noticed that this might not be an appropriate name. For the *function* values we only need N, but here we cache the function values and their gradients (w.r.t. reference and spatial config). On the other hand we have `PointValuesInternal`, which behaves like what I would rather expect from a `FunctionValues` object. If we now argue that `FunctionValues` is a more suitable name as proposed in this PR, why should we stop at the first derivative and not include higher order ones? Would you agree here?

Reply:
Thanks! That is a good point, especially since we have the `function_value` function as well. The question is what would be a suitable name, `FunctionValuesAndGradients` seems a bit verbose?

One way could be to consider what is cached to be an implementation detail. 
If required, we can implement queries for higher-order derivatives (that are not precomputed). Requires the quadrature ref coordinates as well, but at least when called via the `CellValues`, that would be easy. 

But I'm open to other suggestions here!

Reply:
I  really would like to suggest something better, but I am also struggling with the naming name here.